Support `SharingUpdated` web socket notification on patient flow page

nmajor25 opened the pull request:
Shortcut Story ID: [sc-56333]


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- **New Features**
	- Introduced a new message handler for sharing attributes in the model.
  
- **Improvements**
	- Streamlined message handlers to accept a single `attributes` parameter, simplifying state updates.
	- Updated styling for the sharing form to enhance user interface consistency.

- **Tests**
	- Enhanced integration tests for patient flow, including new attributes for actions and improved websocket handling.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

nmajor25 committed:
Support 'SharingUpdated' web socket notification on flow page

nmajor25 commented on src/js/views/patients/sidebar/action/action-sidebar-forms_views.js:
```diff
@@ -58,7 +58,7 @@ function getSharingOpts(sharing) {
 
 const FormSharingView = View.extend({
-  className: 'sidebar__dialog',
+  className: 'sidebar__dialog u-margin--t-24',
```
So that the element has the correct top margin when being shown/hidden on the fly via a web socket notification.

nmajor25 commented on test/integration/patients/patient/patient-flow.js:
I think this is a simple implementation, unless I'm missing something. 

In the action sidebar tests, I did:

1. Start out action as having `sharing: 'disabled'`. i.e. sharing section is hidden in UI.
2. Socket notification comes in that sets `sharing: 'enabled'`. Verify UI elements are now shown.
3. Socket notification comes in tha sets `sharing: 'responded'`. Verify UI elements are updated.

In the action list, I did:

1. Start out action as having `outreach: 'disabled'`. i.e. icon for the action list item is the default version.
2. Socket notification comes in that sets `outreach: 'patient'`. Verify icon for action list item changes. (I'm not sure there's anything else that changes based on that `outreach` attribute).

nmajor25 commented on src/js/entities-service/entities/actions.js:
The notification from the BE contains data for the `recipient`: https://github.com/RoundingWell/care-ops-backend/blob/develop/tests/Notifications/Data/Messages/sharing-updated-message-structure.json.

Can we assume the recipient will always be the patient the action/flow belongs to? And therefore we don't need to do anything on the FE with that recipient data?

paulfalgout replied:
I don't think FE needs to know the recipient.  FE needs to know all of the data such that if the user hard refreshed they'd see the same thing.

paulfalgout commented on src/js/entities-service/entities/actions.js:
```diff
@@ -39,6 +39,9 @@ const _Model = BaseModel.extend({
     DetailsChanged({ details, attributes = {} }) {
       this.set({ details, ...attributes });
     },
+    SharingUpdated({ sharing, outreach, attributes = {} }) {
```
so a bunch of these are wrong.. the only thing doing any work here is attributes

nmajor25 replied:
These have been updated 👍

paulfalgout commented on src/js/views/patients/sidebar/action/action-sidebar-forms_views.js:
```diff
@@ -102,7 +102,7 @@ const FormLayoutView = View.extend({
       <div class="flex-grow" data-form-region></div>
     </div>
-    <div class="flex{{#if hasSharing}} u-margin--t-24{{/if}}">
+    <div class="flex">
```
is the templateContext still in use?

nmajor25 replied:
The `hasSharing` value can be removed. But `hasForm` is still in use in that `templateContext()`.

nmajor25 committed:
Update how ws notification payload data is handled

And update the tests to replicate what actual data structure will look like in prod
